### PR TITLE
fix(cli): Print error message if service does not exist with `trigger delivery`

### LINKED ISSUE #6288: Able to `trigger delivery` with incorrect/non-existing service name
When triggering a sequence e.g. `keptn trigger delivery --project my-project --service my-misspelled-service ...` keptn validates the existence of project `my-project` but is **not** checking whether service `my-misspelled-service` was created.

**To Reproduce**
Trigger a sequence (e.g. delivery) with a service name for which no service was created.

**Expected behavior**
The sequence is not launched.
The user is presented with an error message like it is the case for non-existing projects.

**Current behavior**
The sequence is launched but fails later on, e.g. when the helm service wants to deploy the artifact for the service.

### PR DESCRIPTION
**Pull request by odubajDT**
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

## This PR

- fixed triggered delivery when using non-existing service

### Related Issues

Fixes #6288

### CODE REVIEW

File: cli/cmd/helpers.go [hunk 1]
```diff
@@ -127,3 +129,12 @@ func CheckEndpointStatus(endPoint string) error {
 
 	return nil
 }
+
+func ServiceInSlice(a string, list []*apimodels.Service) bool {
```
Comment (thisthat) [inline cli/cmd/helpers.go:133]:
```suggestion
func ServiceInSlice(service string, serviceList []*apimodels.Service) bool {
```

File: cli/cmd/helpers.go [hunk 2]
```diff
@@ -127,3 +129,12 @@ func CheckEndpointStatus(endPoint string) error {
...
+	for _, b := range list {
```
Comment (thisthat) [inline cli/cmd/helpers.go:134]:
```suggestion
	for _, s := range serviceList {
```

File: cli/cmd/trigger_delivery_test.go [hunk 3]
```diff
@@ -206,3 +206,65 @@ func TestCheckImageNonAvailabilityD(t *testing.T) {
...
+				defer r.Body.Close()
+				w.Write([]byte(metadataMockResponse))
+				return
+			}
+			return
+		}),
+	)
+	defer ts.Close()
+
+	os.Setenv("MOCK_SERVER", ts.URL)
```
Comment (thisthat) [inline cli/cmd/trigger_delivery_test.go:261]: is this necessary?

Comment (odubajDT) [reply cli/cmd/trigger_delivery_test.go:261]: yes, we get errors without it.

Comment (thisthat) [reply cli/cmd/trigger_delivery_test.go:261]: what errors do you get?

Comment (odubajDT) [reply cli/cmd/trigger_delivery_test.go:261]: I will probably re-write the test to be more general :) Hopefully it will be better afterwards :)

Comment (odubajDT) [reply cli/cmd/trigger_delivery_test.go:261]: Yeah and actually for a heads-up: I get segmentation fault regarding some missig variable/nil variable. Did not debug it until now but I can look at it afterwards.

Comment (odubajDT) [reply cli/cmd/trigger_delivery_test.go:267]: I have generalized the test, but I am receiving failures for this test on github, locally works fine. Seems to me like some kind of infrastructure problem, debugging it.

**Review by bacherfl — APPROVED** [review]

**Pull request merged**